model_suggester.py

```diff
@@ -0,0 +1,164 @@
+                success = False
+                while not success: 
+                    try:
+                        output = generate_confounders(
+                            treatment=treatment, 
+                            outcome=outcome, 
+                            variable=variable,
+                            llm=llm)
+
+                        confounder = re.findall(r'<answer>(.*?)</answer>', output['confounders'])
+                        # explanation = re.findall(r'<explanation>(.*?)</explanation>', output['confounders'])
+                        print(confounder)
+                        if confounder[0] == 'A' or confounder[0] == 'A. Yes':
+                            suggested_confounders.append(variable)
+
+                        success = True
+                    
+                    except KeyError:
+                        success = False
+                        continue
+
+        return suggested_confounders
+    
+    def suggest_relationships(self, variables: List[str], llm: guidance.llms):
+
+<<<<<<< HEAD
+        generate_relationships = self._pairwise_relationship_program()
+
+<<<<<<< HEAD
+
+=======
+>>>>>>> modeler
+=======
+        generate_relationships = self.pairwise_relationship_program()
+
+>>>>>>> validator
+        relationships: List[Tuple[str, str]] = []
+        success: bool = False
+
+        for variable_a in variables:
+
+            for variable_b in variables:
+
+                if variable_a != variable_b and (variable_a, variable_b) not in relationships and (variable_b, variable_a) not in relationships:
+                    success = False
+                    while not success: 
+                        try:
+                            output = generate_relationships(variable_a=variable_a, variable_b=variable_b, llm=llm)
+
+                            relationship = re.findall(r'<answer>(.*?)</answer>', output['relationship'])
+                            print(relationship) 
+                            if relationship[0] == 'A' or relationship[0] == 'A. Yes':
+                                relationships.append((variable_a, variable_b))             
+                                
+                            success = True
+                        
+                        except KeyError:
+                            success = False
+                            continue
+        
+        g = nx.DiGraph()
+        g.add_nodes_from(variables)
+        g.add_edges_from(relationships)
+                                
+        return g
+
+<<<<<<< HEAD
+    def _confounder_program(self):
+<<<<<<< HEAD
+
+=======
+>>>>>>> modeler
+=======
+    def confounder_program(self):
+>>>>>>> validator
+        return guidance('''
+        {{#system~}}
+        You are a helpful assistant on causal reasoning. Your goal is to answer questions about cause and effect in arctic sea ice and atmosphere sciences in a factual and concise way.
```

Comment:
let's remove the part of the prompt that is domain specific (arctic sea ice and atmospheric sciences)